Cannot build anymore with 8 BIT samples

### Sonivox version
version: 3.6.16.0

### Describe the bug

Sonivox does not compile anymore after merging #47 when configuring the option USE_16BITS_SAMPLES=OFF (which enables the macro -D_8_BIT_SAMPLES). Build output:

```
arm-wt-22k/lib_src/eas_mdls.c:1501:74: error: passing argument 4 of 'mp3dec_decode_frame' from incompatible pointer type [-Wincompatible-pointer-types]
 1501 |         int samples = mp3dec_decode_frame(&mp3d, convBuf, remainingSize, p, &info);
      |                                                                          ^
      |                                                                          |
      |                                                                          EAS_SAMPLE * {aka signed char *}
```

--- a/.github/workflows/linux-ci.yml
+++ b/.github/workflows/linux-ci.yml
@@ -19,6 +19,7 @@ jobs:
       matrix:
         USE_44KHZ: [ON, OFF]
         USE_16BITS_SAMPLES: [ON, OFF]
+        arch: ["x64", "arm64"]
         include:
           - { icon: '🟥', arch: "x64", os: "ubuntu-24.04" }
           - { icon: '🟩', arch: "arm64", os: "ubuntu-24.04-arm" }

--- a/.github/workflows/linux-ci.yml
+++ b/.github/workflows/linux-ci.yml
@@ -17,6 +17,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        USE_44KHZ: [ON, OFF]
+        USE_16BITS_SAMPLES: [ON, OFF]
         include:
           - { icon: '🟥', arch: "x64", os: "ubuntu-24.04" }
           - { icon: '🟩', arch: "arm64", os: "ubuntu-24.04-arm" }
@@ -33,8 +35,8 @@ jobs:
       run: |
         mkdir -p ${{github.workspace}}/temp
 
-    - name: '${{ matrix.icon }} Configure CMake'
-      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_INSTALL_PREFIX=${{env.INSTALL_LOCATION}}
+    - name: '${{ matrix.icon }} Configure CMake with USE_44KHZ = ${{ matrix.USE_44KHZ }} and USE_16BITS_SAMPLES = ${{ matrix.USE_16BITS_SAMPLES }}'
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_INSTALL_PREFIX=${{env.INSTALL_LOCATION}} -DUSE_44KHZ=${{ matrix.USE_44KHZ }} -DUSE_16BITS_SAMPLES=${{ matrix.USE_16BITS_SAMPLES }}
 
     - name: '${{ matrix.icon }} Build'
       run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}

--- a/arm-wt-22k/lib_src/eas_mdls.c
+++ b/arm-wt-22k/lib_src/eas_mdls.c
@@ -125,10 +125,12 @@
 #include "eas_report.h"
 #include <string.h>
 
+#if defined(_16_BIT_SAMPLES)
 // for mp3 decoding
 #define MINIMP3_IMPLEMENTATION
 #define MINIMP3_ONLY_MP3
 #include "minimp3.h"
+#endif
 
 // for a-law/u-law decoding
 #include "pcm_aulaw.h"
@@ -431,7 +433,9 @@ static EAS_RESULT Parse_wave (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_
 static EAS_RESULT Parse_wsmp (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, S_WSMP_DATA *p);
 static EAS_RESULT Parse_fmt (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, S_WSMP_DATA *p);
 static EAS_RESULT Parse_data (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_I32 size, S_WSMP_DATA *p, EAS_SAMPLE *pSample, EAS_U32 sampleLen);
+#if defined(_16_BIT_SAMPLES)
 static EAS_RESULT Parse_mp3_data (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_I32 size, EAS_SAMPLE *pSample, EAS_I32 *sampleLen);
+#endif
 static EAS_RESULT Parse_lins(SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_I32 size);
 static EAS_RESULT Parse_ins (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_I32 size);
 static EAS_RESULT Parse_insh (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_U32 *pRgnCount, EAS_U32 *pLocale);
@@ -1060,36 +1064,39 @@ static EAS_RESULT Parse_wave (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_
     switch (p->fmtTag)
     {
         case WAVE_FORMAT_MPEGLAYER3:
+#if defined(_16_BIT_SAMPLES)
             if ((result = Parse_mp3_data(pDLSData, dataPos, dataSize, NULL, &size)) != EAS_SUCCESS)
                 return result;
             break;
-    }
+#else
+            return EAS_ERROR_UNRECOGNIZED_FORMAT;
+#endif
+        }
 
-    if (p->loopLength)
-        size += bitDepth / 8;
+        if (p->loopLength)
+            size += bitDepth / 8;
 
-    /* for first pass, add size to wave pool size and return */
-    if (pDLSData->pDLS == NULL)
-    {
-        pDLSData->wavePoolSize += (EAS_U32) size;
+        /* for first pass, add size to wave pool size and return */
+        if (pDLSData->pDLS == NULL) {
+            pDLSData->wavePoolSize += (EAS_U32) size;
+            return EAS_SUCCESS;
+        }
+
+        /* allocate memory and read in the sample data */
+        pSample = (EAS_U8 *) pDLSData->pDLS->pDLSSamples + pDLSData->wavePoolOffset;
+        pDLSData->pDLS->pDLSSampleOffsets[waveIndex] = pDLSData->wavePoolOffset;
+        pDLSData->pDLS->pDLSSampleLen[waveIndex] = (EAS_U32) size;
+        pDLSData->wavePoolOffset += (EAS_U32) size;
+        if (pDLSData->wavePoolOffset > pDLSData->wavePoolSize) {
+            EAS_Report(_EAS_SEVERITY_ERROR, "Wave pool exceeded allocation\n");
+            return EAS_ERROR_SOUND_LIBRARY;
+        }
+
+        if ((result = Parse_data(pDLSData, dataPos, dataSize, p, pSample, (EAS_U32) size))
+            != EAS_SUCCESS)
+            return result;
+
         return EAS_SUCCESS;
-    }
-
-    /* allocate memory and read in the sample data */
-    pSample = (EAS_U8*)pDLSData->pDLS->pDLSSamples + pDLSData->wavePoolOffset;
-    pDLSData->pDLS->pDLSSampleOffsets[waveIndex] = pDLSData->wavePoolOffset;
-    pDLSData->pDLS->pDLSSampleLen[waveIndex] = (EAS_U32) size;
-    pDLSData->wavePoolOffset += (EAS_U32) size;
-    if (pDLSData->wavePoolOffset > pDLSData->wavePoolSize)
-    {
-        EAS_Report(_EAS_SEVERITY_ERROR, "Wave pool exceeded allocation\n");
-        return EAS_ERROR_SOUND_LIBRARY;
-    }
-
-    if ((result = Parse_data(pDLSData, dataPos, dataSize, p, pSample, (EAS_U32)size)) != EAS_SUCCESS)
-        return result;
-
-    return EAS_SUCCESS;
 }
 
 /*----------------------------------------------------------------------------
@@ -1459,6 +1466,7 @@ handle_loop:
  *
  *----------------------------------------------------------------------------
 */
+#if defined(_16_BIT_SAMPLES)
 static EAS_RESULT Parse_mp3_data (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_I32 size, EAS_SAMPLE *pSample, EAS_I32 *sampleLen)
 {
     EAS_RESULT result;
@@ -1526,6 +1534,7 @@ static EAS_RESULT Parse_mp3_data (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, 
 
     return EAS_SUCCESS;
 }
+#endif
 
 /*----------------------------------------------------------------------------
  * Parse_lins ()

